### ISSUE TITLE
support routes stored in Doctrine ORM

### ISSUE THREAD

**Issue by ghost** [symfony-cmf/routing-auto-bundle#19]
This would go along with symfony-cmf/RoutingBundle#83 if it happens.

**Comment by steve-todorov** [comment]
@dantleech @dbu ping? I see this has already been implemented into the RoutingBundle ([PR 122](https://github.com/symfony-cmf/RoutingBundle/pull/122)). Would it be a lot of work to implement it here as well? I'm currently researching the whole Symfony CMF project (including the dynamic routings). If you could give me some guidance I could try to implement it?

**Comment by dantleech** [comment]
No it shouldn't be alot of work. We just need implement a new [Adapter](https://github.com/symfony-cmf/RoutingAutoBundle/blob/master/Adapter/PhpcrOdmAdapter.php) and a new subscriber.

**Comment by WouterJ** [comment]
The difficult thing here is the relation with the content entity. Should we implement this with 2 queries? (one query only getting the AutoRoute and the other the related content, saving both content type and id in the OrmAutoRoute?)

**Comment by fredpeaks** [comment]
Is there any news about the ORM support ?

**Comment by dantleech** [comment]
@fredpeaks I plan to get this into the 2.0 release. I imagine the work to be as follows:

- Refactor the functional tests to support multiple backends, recently started in https://github.com/symfony-cmf/RoutingAutoBundle/pull/173
- Implement a Doctrine ORM adapter, should be easy (tm)
- Make sure the "repair" command in the bundle works with whatever adapter is being used.

**Comment by fredpeaks** [comment]
@dantleech Thank you for the answer, great news !
Do you have some indication on the period where the 2.0 could be released ? I know this can be a tricky question...

**Comment by dantleech** [comment]
Not really. It is next on my list, but it could be a month or two before its started/finished, feel free to start on it if you like however.

**Comment by fredpeaks** [comment]
Ok thanks. I will try to start this based on your PR.

**Comment by fredpeaks** [comment]
For information I've worked on this an the RoutingBundle ORM integration. I've managed to make things work but I've no time for now to finish and create a pull request (there are a lot of tests to write, documentation, refactoring eventually...). I already use this on a production project (not completed now).

If you want to have a look at what I've done here are the repos :
https://github.com/fredpeaks/RoutingBundle
https://github.com/fredpeaks/RoutingAutoBundle

A quick list of what I've done:
 - allow content attached to Route in RoutingBundle (largely inspired by RaindropRoutingBundle)
- allow ORM RedirectRoutes
 - add ORM admin for routes
 - change doctrine mapping and classes hierarchy to allow usage of tables inheritance and  RoutingAutoBundle to create entity mapping instead of RoutingBundle (adding AutoRoute Entity).
- ORM autoroutes

I've been forced to make design choices and there are a lot to discuss about the way things are done.
I've tried to minimize code duplication but the way Doctrine inheritance works forced me to have some. Maybe someone will comme with a better solution.

I'll try to make a pull request next week.
